src/autocomplete/shared/variables.ts

```diff
@@ -0,0 +1,58 @@
+import * as c3 from 'antlr4-c3';
+import {Lexer as LexerType, ParseTree, Parser as ParserType, TokenStream} from 'antlr4ng';
+import {createParser} from './query';
+import {
+    CursorPosition,
+    GetParseTree,
+    LexerConstructor,
+    ParserConstructor,
+    SymbolTableVisitor,
+} from './autocomplete-types';
+import {getScope} from './symbol-table';
+import {computeTokenPosition} from './compute-token-position';
+
+export function getVariablesSuggestions<L extends LexerType, P extends ParserType>(
+    Lexer: LexerConstructor<L>,
+    Parser: ParserConstructor<P>,
+    symbolVariableVisitor: SymbolTableVisitor,
+    getParseTree: GetParseTree<P>,
+    tokenStream: TokenStream,
+    cursor: CursorPosition,
+    query: string,
+): string[] {
+    const parser = createParser(Lexer, Parser, query);
+    const parseTree = getParseTree(parser);
+
+    const tokenPosition = computeTokenPosition(parseTree, tokenStream, cursor);
+
+    if (!tokenPosition) {
+        throw new Error(`Could not find tokenPosition at Ln ${cursor.line}, Col ${cursor.column}`);
+    }
+
+    symbolVariableVisitor.visit(parseTree);
+
+    const symbolTable = symbolVariableVisitor.symbolTable;
```

Comment:
nit: we don't really need this variable, we can just pass `symbolVariableVisitor.symbolTable` directly to `suggestVariablesFunction`